src/hooks/useContentProtection.ts

```diff
@@ -24,21 +24,20 @@ const useContentProtection = <T>(
   }));
   const host = signingConfig?.host;
   const drmPolicyId = contentProtection?.drm?.defaultPolicyId ?? signingConfig?.drmPolicyId;
-  const drmEnabled = !!drmPolicyId;
-  const signingEnabled = !!host || !!urlSigning;
+  const signingEnabled = !!urlSigning;
```

Comment:
```suggestion
  const signingEnabled = !!urlSigning || !!host || (!!drmPolicyId && !host);
```

Reply:
why is host here? This way we will need to include `signingEnabled` in the other providers condition and it will be true if jw integration is present as well.

Reply:
If host is set for the legacy config, it means signing is enabled (there was not a separate custom param)

Reply:
then we should change the other providers condition to be like: `if (!!id && !!host && signingEnabled)`, otherwise it will always return the old signing method.